protocol/linkpreview_unfurler_status.go

```diff
@@ -0,0 +1,195 @@
+package protocol
+
+import (
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/api/multiformat"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/images"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/communities"
+)
+
+type StatusUnfurler struct {
+	m      *Messenger
+	logger *zap.Logger
+	url    string
+}
+
+func NewStatusUnfurler(URL string, messenger *Messenger, logger *zap.Logger) *StatusUnfurler {
+	return &StatusUnfurler{
+		m:      messenger,
+		logger: logger,
+		url:    URL,
+	}
+}
+
+func buildThumbnail(image *images.IdentityImage, thumbnail *common.LinkPreviewThumbnail) error {
+	if image.IsEmpty() {
+		return nil
+	}
+
+	var err error
+
+	thumbnail.Width, thumbnail.Height, err = images.GetImageDimensions(image.Payload)
+	if err != nil {
+		return fmt.Errorf("failed to get image dimensions: %w", err)
+	}
+
+	thumbnail.DataURI, err = image.GetDataURI()
+	if err != nil {
+		return fmt.Errorf("failed to get data uri: %w", err)
+	}
+
+	return nil
+}
+
+func (u *StatusUnfurler) buildContactData(contactData *ContactURLData) (*common.StatusContactLinkPreview, error) {
+
+	contactID, err := multiformat.DeserializeCompressedKey(contactData.PublicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	c := new(common.StatusContactLinkPreview)
+	c.PublicKey = contactData.PublicKey
+	c.DisplayName = contactData.DisplayName
+	c.Description = contactData.Description
+
+	contact := u.m.GetContactByID(contactID)
+	if contact == nil {
+		return c, nil
+	}
+
+	// TODO: Should we try to fetch from waku?
+	//if contact == nil {
```

Comment:
Sure, I'll drop it.
I've added a question about this to the PR anyway. And you remember I also discussed this with you, it will probably require more app-backend signals communication, so we're not going to do this now I think.

Reply:
Aaaand now this code will be uncommented 😄